libgraph/src/analytics/k_shortest_paths/ksssp.cpp

```diff
@@ -0,0 +1,420 @@
+/*
+ * This file belongs to the Galois project, a C++ library for exploiting
+ * parallelism. The code is being released under the terms of the 3-Clause BSD
+ * License (a copy is located in LICENSE.txt at the top-level directory).
+ *
+ * Copyright (C) 2018, The University of Texas at Austin. All rights reserved.
+ * UNIVERSITY EXPRESSLY DISCLAIMS ANY AND ALL WARRANTIES CONCERNING THIS
+ * SOFTWARE AND DOCUMENTATION, INCLUDING ANY WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR ANY PARTICULAR PURPOSE, NON-INFRINGEMENT AND WARRANTIES OF
+ * PERFORMANCE, AND ANY WARRANTY THAT MIGHT OTHERWISE ARISE FROM COURSE OF
+ * DEALING OR USAGE OF TRADE.  NO WARRANTY IS EITHER EXPRESS OR IMPLIED WITH
+ * RESPECT TO THE USE OF THE SOFTWARE OR DOCUMENTATION. Under no circumstances
+ * shall University be liable for incidental, special, indirect, direct or
+ * consequential damages or loss of profits, interruption of business, or
+ * related expenses which may arise from use of Software or Documentation,
+ * including but not limited to those resulting from defects in Software and/or
+ * Documentation, or loss or inaccuracy of data of any kind.
+ */
+
+#include "katana/analytics/k_shortest_paths/ksssp.h"
+
+#include "katana/AtomicHelpers.h"
+#include "katana/Reduction.h"
+#include "katana/Statistics.h"
+#include "katana/TypedPropertyGraph.h"
+#include "katana/analytics/BfsSsspImplementationBase.h"
+#include "katana/analytics/kSsspImplementationBase.h"
+#include "katana/gstl.h"
+
+using namespace katana::analytics;
+
+struct Path {
+  uint32_t parent;
```

Comment:
Can we use typedef for representing parents type so that we can use same type throughout and can easily change it?

Reply:
Would you prefer for this struct to be moved to a different file?

Reply:
no, he just want to define something like `typedef uint32_t Parent`; and then use `parent` like this: `Parent parent`.